daml-lf/spec/daml-lf-1.rst

```diff
@@ -1013,6 +1022,46 @@ We now formally defined *well-formed types*. ::
    ————————————————————————————————————————————— TyScenario
      Γ  ⊢  'Scenario' : ⋆ → ⋆
 
+   ————————————————————————————————————————————— TyAnyException
+     Γ  ⊢  'AnyException' : ⋆
+
+   ————————————————————————————————————————————— TyUserException
+     Γ  ⊢  'UserException' : ⋆
+
+   ————————————————————————————————————————————— TyUserException
+     Γ  ⊢  'ArithmeticException' : ⋆
+
+   ————————————————————————————————————————————— TyPreconditionException
+     Γ  ⊢  'PreconditionException' : ⋆
+
+
+Exception types
+...............
+
+To state the typing rules related to exception handling, we need the notion of
+*exception types*. As the name suggests, values of these types are the ones that
+can be thrown and caught by the exception handling mechanism. ::
+
+                      ┌────────┐
+  Exception types     │ ⊢ₑ  τ  │
+                      └────────┘
+
+    ———————————————————————————————————————————————————————————————— ETyUserException
+      ⊢ₑ  'UserException'
+
+    ———————————————————————————————————————————————————————————————— ETyArithmeticException
+      ⊢ₑ  'ArithmeticException'
+
+    ———————————————————————————————————————————————————————————————— ETyPreconditionException
+      ⊢ₑ  'PreconditionException'
+
+      'exception' (x : T) ↦ …  ∈  〚Ξ〛Mod
+    ———————————————————————————————————————————————————————————————— ETyDefException
+      ⊢ₑ  Mod:T
+
+Note that ``'AnyException'`` is not an exception type in order to avoid having
+``'AnyException'`` wrapped into ``'AnyException'``.
```

Comment:
I don't feel strongly about this, but we could consider `AnyException` an exception type, and avoid rewrapping it at runtime when calling `to_any_exception` and `from_any_exception`.

Reply:
I'm inclined to avoid the special case in the interpreter for now but would be willing to add it later if it turns out to be necessary.

Reply:
Good point. I think it really won't make a difference, damlc will only use the `to/from` primitives on user-defined and other primitive exceptions anyway.